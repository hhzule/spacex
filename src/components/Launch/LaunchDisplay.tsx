import React from "react";

const LaunchDisplay = ({ name, year, status }: any) => {
  return (
    <div>
      <div className={status ? "red" : "green"}>
        <div className="wrapper">
          <p>
            {" "}
            {name}
            <span></span>
            {year}
          </p>
          <div className="learn">
            {status ? <p>Success</p> : <p>Failure</p>}
          </div>
        </div>
      </div>
    </div>
  );
};

export default LaunchDisplay;
