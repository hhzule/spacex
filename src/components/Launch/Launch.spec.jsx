import React from "react";
import { shallow, render } from "enzyme";
import LaunchDisplay from "./LaunchDisplay";
// import { LaunchesQuery } from "../../generated/graphql";
// import { useLaunchesQuery } from "../../generated/graphql";
// import { MockedProvider } from "@apollo/client/testing";

// const mocks = [
//   {
//     request: {
//       query: useLaunchesQuery,
//     },
//     result: {
//       data: {
//         launches: {
//           mission_name: "first",
//           details: "starlink launch",
//           launch_year: 2020,
//           launch_success: true,
//         },
//       },
//     },
//   },
// ];

// const mock = {
//   launches: {
//     mission_name: "first",
//     details: "starlink launch",
//     launch_year: 2020,
//     launch_success: true,
//   },
// };

// const component = render(
//   <MockedProvider mocks={mocks} addTypename={false}>
//     <Launch />
//   </MockedProvider>
// );
// test("renders the main div", () => {
//   console.log(component, " componenet");
//   expect(component).toBeTruthy();
// });

const component = shallow(
  <LaunchDisplay name={"star"} status={true} year={2020} />
);
test("renders the main div", () => {
  expect(component).toMatchSnapshot();
});
