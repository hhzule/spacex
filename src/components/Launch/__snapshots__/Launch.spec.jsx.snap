// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the main div 1`] = `
<div>
  <div
    className="red"
  >
    <div
      className="wrapper"
    >
      <p>
         
        star
        <span />
        2020
      </p>
      <div
        className="learn"
      >
        <p>
          Success
        </p>
      </div>
    </div>
  </div>
</div>
`;
